### ISSUE TITLE
problem to check type data in dataclass

### ISSUE THREAD

**Issue by dooyeoung** [microsoft/pyright#2743]
**Describe the bug**
After importing the `uuid` module
I declared a variable with the name `uuid` in dataclass.
If typing `uuid` is used in other variable declarations, it is not recognized as an imported module `uuid`.
Below is the error content.
```python
error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)
```

**Screenshots or Code**
```python
import uuid

@dataclass
class DataObject:
    name: str
    uuid: uuid.UUID
    active: bool
    parent_uuid: typing.Optional[uuid.UUID] = None
```

**VS Code extension or command-line**
pyright 1.1.198

**Comment by JelleZijlstra** [comment]
Pyright is correct there: within the class the name `uuid` refers to the class attribute, not to the module. To work around this I'd recommend writing `import uuid as _uuid` and referring to the type as `_uuid.UUID`.

**Comment by erictraut** [comment]
As @JelleZijlstra said, I think pyright is doing the correct thing here.

In addition to the suggested solution provided above, here are a few others:
1. Enclose the `uuid.UUID` type annotation in quotes. This will tell a type checker to use forward declaration rules for resolving the symbol, and it will use the `uuid` symbol in the global scope rather than in the class scope.
2. Add `from __future__ import annotations` to the top of the file.
